torch/csrc/jit/interpreter.cpp

```diff
@@ -193,6 +362,55 @@ struct CodeImpl {
   std::vector<bool> bool_data;
 };
 
+
+struct Registers {
+  Registers(size_t size)
+  : registers(size) {}
+  Registers(const Registers & rhs)
+  : registers(rhs.registers) {
+    for(auto & r : registers) {
+      if(isValid(r))
+        r->retain();
+    }
+  }
+  ~Registers() {
+    for(auto & r : registers) {
+      if(isValid(r))
+        r->release();
+    }
+  }
+  at::RefCounted*& operator[](size_t i) {
+    return registers[i];
+  }
+  //guarenteed to be non-null
+  at::RefCounted* release(size_t i) {
+    auto & v = registers[i];
+    if(isValid(v)) {
+      v->release();
+      v = nullptr;
+    }
+    return v;
+  }
+  void save(size_t i, const at::Tensor & t) {
+    registers[i] = at::Tensor(t).detach();
+  }
+  at::Tensor load(size_t i, bool andFree) {
+    auto & v = registers[i];
+    if(andFree) {
+      auto r = at::Tensor(static_cast<at::TensorImpl*>(v), false);
```

Comment:
Now, that's an `unsafeTensorBorrow` ;)

Reply:
per pybind11:

```
at::Tensor(static_cast<at::TensorImpl*>(v), false) <-- reinterpret_steal
at::Tensor(static_cast<at::TensorImpl*>(v), true) <-- reinterpret_borrow
```

Pybind11's terminology however, is confusing, so I am going to change the names and just be more explicit.